src/main/java/ui/BoardAutoCreator.java

```diff
@@ -74,6 +86,46 @@ private void createMilestoneBoard() {
                         "It is saved under the name \"" + boardName + "\".");
     }
 
+    private void createWorkAllocationBoard() {
+        logger.info("Creating " + WORK_ALLOCATION + " board");
+
+        List<TurboUser> userList = ui.logic.getRepo(ui.logic.getDefaultRepo()).getUsers();
+
+        if (userList.isEmpty()) {
+            DialogMessage.showInformationDialog("Auto-create Board - " + WORK_ALLOCATION,
+                    WORK_ALLOCATION + " board cannot be created as your" +
+                            " account has no push access to the default repo.");
+
+            return;
+        }
+
+        panelControl.closeAllPanels();
+
+        FilterPanel panelAdded;
+        TurboUser currUser;
+        String realName, loginName;
+
+        for (int i = 0; i < Math.min(MAX_WORK_ALLOCATION_PANELS, userList.size()); i++){
+            currUser = userList.get(i);
+            realName = currUser.getRealName();
+            loginName = currUser.getLoginName();
+            panelAdded = panelControl.addPanelAt(i);
+            panelAdded.setPanelName("Work allocated to " + (!realName.isEmpty() ? realName : loginName));
+            panelAdded.setFilterByString("assignee:" + loginName + " sort:milestone,status");
+        }
+
+        panelControl.selectFirstPanel();
+
+        String boardName = Utility.generateName(WORK_ALLOCATION,
+                prefs.getAllBoards().keySet().stream().collect(Collectors.toList()));
+
+        triggerBoardSaveEventSequence(boardName);
+
+        DialogMessage.showInformationDialog("Auto-create Board - " + WORK_ALLOCATION,
+                WORK_ALLOCATION + " board has been created and loaded.\n\n" +
+                        "It is saved under the name \"" + boardName + "\".");
```

Comment:
Something like this? It can be modified later when #1074 is done
```java
//panelData = pairs of filter and panel name
private void createBoard(List<Pair<String,String>> panelData, String boardName){
        //everything to do with board creation
}

```